### PR TITLE
db layer  added

### PR DESCRIPTION
**Pull request by Asifkhan-akhan**
database layer added

### CODE REVIEW

File: db/mongo/client.go [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+	log().Infof("initializing mongodb: %s", uri)
+	cli, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to db")
+	}
+	return &client{conn: cli}, nil
+}
+
+func (c *client) CreateOrUpdateBooking(booking *models.Booking) (int, error) {
+	if booking.ID == 0 {
```
Comment (OvaisTariq95) [inline db/mongo/client.go:46]: should be nil not 0

File: db/mongo/client.go [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+// NewClient initializes a mysql database connection
+func NewClient() (db.DataStore, error) {
+	uri := fmt.Sprintf("mongodb://%s:%s", viper.GetString(config.DbHost), viper.GetString(config.DbPort))
+	log().Infof("initializing mongodb: %s", uri)
+	cli, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to db")
+	}
+	return &client{conn: cli}, nil
```
Comment (OvaisTariq95) [inline db/mongo/client.go:42]: newline before

**Review by OvaisTariq95 — CHANGES_REQUESTED** [review]

File: db/mongo/client.go (hunk 1 above)
Comment (Asifkhan-akhan) [reply db/mongo/client.go:46]: I have configured the ID field in my model as an int, which is not a pointer type( it holds value directly) and a non-pointer type cannot be checked with nil in Go

File: db/mongo/client.go (hunk 2 above)
Comment (Asifkhan-akhan) [reply db/mongo/client.go:42]: changes done and pushed

**Review by OvaisTariq95 — APPROVED** [review]

**Pull request merged**